windows/src/desktop/kmshell/install/UfrmInstallKeyboard.pas

```diff
@@ -161,9 +185,29 @@ function InstallFiles(Owner: TComponent; const FileNames: TWideStrings; ASilent:
     try
       kmcom.Keyboards.Refresh;
 
-      if AnsiSameText(ExtractFileExt(FileNames[i]), '.kmp')
-        then kmcom.Packages.Install(FileNames[i], True)
-        else kmcom.Keyboards.Install(FileNames[i], True);
+      FilenameBCP47 := Filenames[I].Split(['=']);
+      Filename := FilenameBCP47[0];
+      IsPackage := AnsiSameText(ExtractFileExt(FileNames[i]), '.kmp');
```

Comment:
I take it Delphi is case-insensitive?  I don't see a declaration for `i`, but I do for `I`.  If so, it's probably still better that we case variables consistently.

Since `i` is used through much of the rest of the function, perhaps it's the `I` declaration (line 183) that should shift.  Either way.

Reply:
Yes, Delphi is case-insensitive.